### ISSUE TITLE
`sl status` is too slow for shell prompts

### ISSUE THREAD

**Issue by ammario** [facebook/sapling#708]
I'm trying to use `sl status` in my shell prompt, but the command consistently takes 400ms whereas `git status` in the same repo takes 70ms.

**Comment by vegerot** [comment]
agreed.  It's better just to read the `.sl/` directory directly.  This is what we do in our [zsh and bash prompt](https://github.com/facebook/sapling/blob/9c9617c0aac2682c66b4352521de7169e6dbf1c2/eden/scm/contrib/scm-prompt.sh)

**Comment by ammario** [comment]
Thanks for sharing.

Unfortunately it seems like you still need `sl status` to get the dirty state (which is very important to me), so the file system tricks only shaves ~100ms (30%) off.
